frontend/src/pages/GraphicalSelector/CourseInfo/GraphicalPlannerButton.tsx

```diff
@@ -0,0 +1,95 @@
+import React, { FunctionComponent, useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { PlusOutlined, StopOutlined } from '@ant-design/icons';
+import { Button } from 'antd';
+import axios from 'axios';
+import { Course, UnselectCourses } from 'types/api';
+import { PlannerCourse } from 'types/planner';
+import prepareUserPayload from 'utils/prepareUserPayload';
+import type { RootState } from 'config/store';
+import { addToUnplanned, removeCourses } from 'reducers/plannerSlice';
+
+interface PlannerButtonCodeProps {
+  course: Course;
+  onChange?: () => void;
+}
+
+const PlannerButtonCode: FunctionComponent<PlannerButtonCodeProps> = ({ course, onChange }) => {
+  const coursesInPlanner = useSelector((state: RootState) => state.planner.courses);
+  const { degree, planner } = useSelector((state: RootState) => state);
+
+  const id = course.code;
+  const dispatch = useDispatch();
+  const [isAddedInPlanner, setIsAddedInPlanner] = useState(!!coursesInPlanner[id]);
+  const [loading, setLoading] = useState(false);
+
+  const addCourseToPlannerTimeout = (isCourseInPlanner: boolean) => {
```

Comment:
this whole file seems like it was stolen wholecloth from courseSelector? Dont copy, please factor them into a common component instead.